src/lua/api-gateway/util/OauthClient.lua

```diff
@@ -16,6 +16,52 @@ function OauthClient:new(o)
     return o
 end
 
+--- Loads a lua gracefully. If the module doesn't exist the exception is caught, logged and the execution continues
+-- @param module path to the module to be loaded
+--
+local function loadrequire(module)
+    ngx.log(ngx.DEBUG, "Loading module [" .. tostring(module) .. "]")
+    local function requiref(module)
+        require(module)
+    end
+
+    local res = pcall(requiref, module)
+    if not (res) then
+        ngx.log(ngx.WARN, "Could not load module [", module, "].")
+        return nil
+    end
+    return require(module)
+end
+
+local restyDogstatsd = loadrequire('resty_dogstatsd')
+
+--- Returns an instance of dogstatsd only if it does not already exist
+function OauthClient:getDogstatsd()
```

Comment:
this method has no sense here - would put in a dogstatsd wrapper that treats lazy require as well